### PR TITLE
[baresip-libre] update to 3.4.0

### PR DESCRIPTION
**Pull request by alfredh**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/baresip-libre/portfile.cmake [hunk 1]
```diff
@@ -6,7 +6,7 @@ vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO baresip/re
     REF "v${VERSION}"
-    SHA512 9813f66e73df80e00f1947c8c2af0f9dc6afe8d197a453b622458f80d522edd5b420bfc783161d311b36936947764a7a1733038c121210ab65f8621a3502d372
+    SHA512 97ea35e4d4f36a9b3e47bb942497d495247e01106bcceef98fba4ab8e36061acaca88d12568990f9c8014b1061941ea5e2c6e8c0287e6dcd83a60a70400b083d
     HEAD_REF main
     PATCHES
         wip.patch
```
Comment (dg0yt) [inline ports/baresip-libre/portfile.cmake:12]: I really would like to see any "wip" go away before a PR changes to "Ready for review".

**Review by BillyONeal — DISMISSED** [review]
I agree with @dg0yt that the patch name should match what it is doing. (If it was just a version bump with no other edits I would be OK with leaving it alone but you did edit the patch)

I would just push a rename myself for you but I don't have a good enough understanding of what the patch is doing to name it correctly.

**Comment by alfredh** [comment]
I have renamed the patch to "cmake.patch". My plan is to merge this patch in the upstream
project, so that this patch files goes away.

Next patches coming from me:

1. update to version 3.5.0

**Comment by alfredh** [comment]
please note, I did not modify that patch on purpose. It has mainly been rebased
from 3.3.0 to 3.4.0

I have renamed the patch file again, hopefully this name is better.

That patch file has been written by @dg0yt btw.
If you have suggestions for better name, please let me know.

The patch should be ready to merge now.

**Comment by zkWildfire** [comment]
@alfredh I'm still not a fan of the new name. The goal of renaming the patch file should be to tell me what the patch does. `use-c11.patch` tells me exactly what effects applying the patch has. The name `cmake-fixes.patch` is an improvement but I'd still like to see it become less generic.

Based on a quick look at the changes, it looks like there may be multiple changes that are independent of each other? For example, switching `-DVERSION` to `-DRE_VERSION` probably doesn't also require the change to the `target_link_libraries()` call. Ideally, what I'd like to see is a set of patch files that individually targets changes that are necessary to make libre play nice with other libraries, or whatever the purpose behind the changes is.

I can't say for sure whether these names would be accurate for the changes so I'd highly recommend you change them as necessary, but these are along the lines of what I'd like to see for the patch file's name and how it could be split up:
* `scope-version-define.patch`
  * For the `VERSION` -> `RE_VERSION` change
* `hide-libre-link-libraries.patch`
  * For adding `PRIVATE` to the `target_link_libraries()` call
* `fix-static-library-build.patch` (this is also a generic name, but I don't know enough about what the changes are solving to propose a more specific name)
  * For the static library specific changes

**Comment by dg0yt** [comment]
> That patch file has been written by @dg0yt btw.

It was a collection of suggestions for *upstream changes*, IIRC from a branch called `wip`. It wasn't meant to end up in the port as a single patch.

**Comment by Cheney-W** [comment]
@alfredh Would you be willing to split the patch according to zkWildfire's [suggestion](https://github.com/microsoft/vcpkg/pull/34225#issuecomment-1749254224)?

**Comment by Cheney-W** [comment]
I have converted this PR to draft status. When you respond, please revert to "ready for review".

**Comment by alfredh** [comment]
please note:

this change: -DVERSION to -DRE_VERSION 

is a change from 3.3.0 to 3.4.0 -- it is not part of the patch.

we have already merged one patch to upstream main:

https://github.com/baresip/re/commit/aeff9e5d158a15b4ae9ef14148483528d845b467

we are also planning to merge

**Comment by alfredh** [comment]
there are parts of the patch in one file which depends on parts in the other file:

For example:

@PC_LIBNAME

is defined in CMakeLists.txt and used in libre.pc.in

Therefore it makes sense to keep this patch as one file.

**Comment by dg0yt** [comment]
@zkWildfire The key issue with the static linkage is *to export* the link libraries, in pc files and cmake config.

**Review by Cheney-W — APPROVED** [review]

**Pull request merged**